antarest/study/storage/variantstudy/model/command/generate_thermal_cluster_timeseries.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import shutil
+import tempfile
+import typing as t
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+from antares.tsgen.duration_generator import ProbabilityLaw
+from antares.tsgen.random_generator import MersenneTwisterRNG
+from antares.tsgen.ts_generator import ThermalCluster, ThermalDataGenerator
+
+from antarest.study.storage.rawstudy.model.filesystem.config.model import Area, FileStudyTreeConfig
+from antarest.study.storage.rawstudy.model.filesystem.config.thermal import LocalTSGenerationBehavior
+from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
+from antarest.study.storage.variantstudy.model.command.common import CommandName, CommandOutput
+from antarest.study.storage.variantstudy.model.command.icommand import ICommand, OutputTuple
+from antarest.study.storage.variantstudy.model.model import CommandDTO
+
+logger = logging.getLogger(__name__)
+
+
+class GenerateThermalClusterTimeSeries(ICommand):
+    """
+    Command used to generate thermal cluster timeseries for an entire study
+    """
+
+    command_name = CommandName.GENERATE_THERMAL_CLUSTER_TIMESERIES
+    version = 1
+
+    _INNER_MATRICES: t.List[str] = []
+
+    def _apply_config(self, study_data: FileStudyTreeConfig) -> OutputTuple:
+        return CommandOutput(status=True, message="Nothing to do"), {}
+
+    def _apply(self, study_data: FileStudy) -> CommandOutput:
+        study_path = study_data.config.study_path
+        with tempfile.TemporaryDirectory(
+            suffix=".thermal_timeseries_gen.tmp", prefix="~", dir=study_path.parent
+        ) as path:
+            tmp_dir = Path(path)
+            try:
+                shutil.copytree(study_path / "input" / "thermal" / "series", tmp_dir, dirs_exist_ok=True)
+                self._build_timeseries(study_data, tmp_dir)
+            except Exception as e:
+                logger.error(f"Unhandled exception when trying to generate thermal timeseries: {e}", exc_info=True)
+                raise
+            else:
+                self._replace_safely_original_files(study_path, tmp_dir)
+                return CommandOutput(status=True, message="All time series were generated successfully")
+
+    def _build_timeseries(self, study_data: FileStudy, tmp_path: Path) -> None:
+        # 1- Get the seed and nb_years to generate
+        # NB: Default seed in IHM Legacy: 5489, default seed in web: 3005489.
+        general_data = study_data.tree.get(["settings", "generaldata"], depth=3)
+        thermal_seed = general_data["seeds - Mersenne Twister"]["seed-tsgen-thermal"]
+        nb_years = general_data["general"]["nbtimeseriesthermal"]
+        # 2 - Build the generator
+        rng = MersenneTwisterRNG(seed=thermal_seed)
+        generator = ThermalDataGenerator(rng=rng, days=365)
```

Comment:
I didn't understand it at first but since we always want to generate 8760 rows (Thibault says), we need to have 365 days (*24 inside the generation package).